src/database/DataTypes/ItemBatch.js

```diff
@@ -125,6 +125,31 @@ export class ItemBatch extends Realm.Object {
   toString() {
     return `${this.itemName} - Batch ${this.batch}`;
   }
+
+  /**
+   * Returns all SensorLog objects related to this ItemBatch
+   * which has recorded a breach.
+   */
+  get breaches() {
+    return this.sensorLogs.filtered('isInBreach = $0', true);
+  }
+
+  /**
+   * Returns if this ItemBatch has been in a breach.
+   */
+  get hasBreached() {
+    return this.breaches.length > 0;
+  }
+
+  /**
+   * Returns an object {maxTemperature, minTemperature} for all
+   * recorded temperatures for this ItemBatch.
+   */
+  get temperatureExposure() {
+    const maxTemperature = this.sensorLogs.max('temperature');
```

Comment:
what would this return if nothing in sensorLogs ?

Reply:
hmm, it returns undefined. I have set the default case to be  `{ maxTemperature: -Infinity, minTemperature: Infinity }` as above.